convert2rhel/systeminfo.py

```diff
@@ -305,6 +318,52 @@ def get_enabled_rhel_repos(self):
         #     )
         return self.submgr_enabled_repos if not tool_opts.no_rhsm else tool_opts.enablerepo
 
+    def _check_internet_access(self, host="8.8.8.8", port=53, timeout=3):
+        """Check whether or not the machine is connected to the internet.
+
+        This method will try to estabilish a socket connection through the
+        default host in the method signature (8.8.8.8). If it's connected
+        successfully, then we know that internet is accessible from the host.
+
+        .. warnings::
+            We might have some problems with this if the host machine is using
+            a NAT gateway to route the outbound requests
+
+        .. seealso::
+            * Comparison of different methods to check internet connections: https://stackoverflow.com/a/33117579
+            * Redirecting IP addresses: https://stackoverflow.com/a/33117579https://superuser.com/questions/954665/how-to-redirect-route-ip-address-to-another-ip-address
+
+        :param host: The host to establish a connection to. Defaults to "8.8.8.8".
+        :type host: str
+        :param port: The port for the connection. Defaults to 53.
+        :type port: int
+        :param timeout: The timeout in seconds for the connection. Defaults to 3.
+        :type port: int
+        :return: Return boolean indicating whether or not we have internet access.
+        :rtype: bool
+        """
+        try:
+            self.logger.info("Checking internet connectivity using host '%s' and port '%s'." % (host, port))
+            socket.setdefaulttimeout(timeout)
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((host, port))
+            return True
+        except OSError:
+            self.logger.warning("Couldn't connect to the host '%s'." % host)
+            return False
+
+    def corresponds_to_rhel_eus_release(self):
+        """Return whether the current minor version corresponds to a RHEL Extended Update Support (EUS) release.
+
+        For example if we detect CentOS Linux 8.4, the corresponding RHEL 8.4 is an EUS release, but if we detect
+        CentOS Linux 8.5, the corresponding RHEL 8.5 is not an EUS release.
+
+        :return: Whether or not the current system has a EUS correspondent in RHEL.
+        :rtype: bool
+        """
+        system_version = "%s.%s" % (self.version.major, self.version.minor)
+        EUS_MINOR_VERSIONS = ["8.4", "8.6", "8.8", "8.10"]
```

Comment:
```suggestion
        EUS_MINOR_VERSIONS = ["8.4", "8.6", "8.8"]
```
I've now realized that the last minor version won't be an EUS minor version: https://access.redhat.com/articles/rhel-eus#c5.

Reply:
done.